src/Network/NWBrowser.cs

```diff
@@ -113,22 +117,65 @@ public NWParameters Parameters
 		[DllImport (Constants.NetworkLibrary)]
 		unsafe static extern void nw_browser_set_browse_results_changed_handler (OS_nw_browser browser, void *handler);
 
-		delegate void nw_browser_browse_results_changed_handler_t (IntPtr block, IntPtr oldResult, IntPtr newResult);
+		delegate void nw_browser_browse_results_changed_handler_t (IntPtr block, IntPtr oldResult, IntPtr newResult, bool completed);
 		static nw_browser_browse_results_changed_handler_t static_ChangesHandler = TrampolineChangesHandler;
 
 		[MonoPInvokeCallback (typeof (nw_browser_browse_results_changed_handler_t))]
-		static void TrampolineChangesHandler (IntPtr block, IntPtr oldResult, IntPtr newResult)
+		static void TrampolineChangesHandler (IntPtr block, IntPtr oldResult, IntPtr newResult, bool completed)
 		{
-			var del = BlockLiteral.GetTarget<Action<NWBrowseResult, NWBrowseResult>> (block);
+			var del = BlockLiteral.GetTarget<NWBrowserChangesDelegate> (block);
 			if (del != null) {
-				using (var nwOldResult = new NWBrowseResult (oldResult, owns: false))
-				using (var nwNewResult = new NWBrowseResult (newResult, owns: false))
-					del (nwOldResult, nwNewResult);
+				// we do the cleanup of the objs in the internal handlers
+				var nwOldResult = new NWBrowseResult (oldResult, owns: false);
+				var nwNewResult = new NWBrowseResult (newResult, owns: false);
+				del (nwOldResult, nwNewResult, completed);
+			}
+		}
+
+		public delegate void NWBrowserChangesDelegate (NWBrowseResult oldResult, NWBrowseResult newResult, bool completed);
```

Comment:
it's harder to find as an inner type

Reply:
You mean move it outside the class, fair enough